### PR TITLE
Dp problem

### PR DESCRIPTION
**Pull request by Abby3017**
Added solutions for two dp problem.

1. Longest common subsequence
2. Maximum coin change way for given number

### CODE REVIEW

File: dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package main
+
```
Comment (addy1997) [inline dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go:2]: please eliminate the blank lines

File: dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"reflect"
+	"testing"
+)
+
```
Comment (addy1997) [inline dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go:7]: eliminate blank lines

File: dynamic_programming/number_coin_change/number_coin_change.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package main
```
Comment (addy1997) [inline dynamic_programming/number_coin_change/number_coin_change.go:1]: please add a small description about the code

File: dynamic_programming/number_coin_change/number_coin_change_test.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package main
```
Comment (addy1997) [inline dynamic_programming/number_coin_change/number_coin_change_test.go:1]: add small description about the code at the top.

File: dynamic_programming/number_coin_change/number_coin_change.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (addy1997) [inline dynamic_programming/number_coin_change/number_coin_change.go:2]: add a small description of the code at the top.

**Comment by addy1997** [comment]
Hi @Abby3017, 

Thanks for your contributions. I have made a few suggestions.

File: dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go (hunk 1 above)
Comment (Abby3017) [reply dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go:2]: There should be blank lines between package name and import. I can't find official documentation for now but I am attaching the digitalOcean tutorials. You can check the format of Go code.
https://www.digitalocean.com/community/tutorials/importing-packages-in-go

File: dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go (hunk 2 above)
Comment (Abby3017) [reply dynamic_programming/longest_common_subsequence/longest_common_subsequence_test.go:7]: between function name and import blank line should be there. See the above comment for this too

File: dynamic_programming/number_coin_change/number_coin_change_test.go (hunk 4 above)
Comment (Abby3017) [reply dynamic_programming/number_coin_change/number_coin_change_test.go:1]: For test file, what kind of description you need, it standard format to check output of any func. Can you elaborate on it ?

File: dynamic_programming/number_coin_change/number_coin_change.go (hunk 5 above)
Comment (Abby3017) [reply dynamic_programming/number_coin_change/number_coin_change.go:2]: duplicate of other comment. We can close it.

File: dynamic_programming/number_coin_change/number_coin_change_test.go (hunk 4 above)
Comment (addy1997) [reply dynamic_programming/number_coin_change/number_coin_change_test.go:1]: With description, I meant what the code does or what a particular program is about. For eg: line like **program to check the palindrome**

**Comment by addy1997** [comment]
> Hi @Abby3017,
> 
> Thanks for your contributions. I have made a few suggestions.

Please refer this [text](https://www.golang-book.com/books/intro/2#section1) for the blank lines issue.
`
The next line is a blank line. Computers represent newlines with a special character (or several characters). Newlines, spaces and tabs are known as whitespace (because you can't see them). Go mostly doesn't care about whitespace, we use it to make programs easier to read. (You could remove this line and the program would behave in exactly the same way)`

Now onwards, please add the description atop the main file. 
I am merging it.

**Comment by Abby3017** [comment]
Just to clarify, description should be added before the package name ?

newline is there for code readability and we want people who see this repo to be able to easily understand the code for given problem rather than struggle for readability as we tried to save few lines.
For me a good format code makes more sense as it is easy for eyes to follow.
Thanks for your link.

**Pull request merged**